hazelcast/src/main/java/com/hazelcast/query/impl/predicates/BetweenPredicate.java

```diff
@@ -31,8 +31,10 @@
  */
 public class BetweenPredicate extends AbstractIndexAwarePredicate {
 
-    Comparable to;
-    Comparable from;
+    protected Comparable from;
+    protected Comparable to;
+    private transient volatile Comparable fromConverted;
```

Comment:
The predicate instance is shared by partitions on each node. Thus if one initialises the other ones should see it properly.